features/include/pcl/features/impl/multiscale_feature_persistence.hpp

```diff
@@ -195,6 +194,7 @@ pcl::MultiscaleFeaturePersistence<PointSource, PointFeature>::extractUniqueFeatu
     }
     unique_features_indices_.emplace_back (std::move(indices_per_scale));
     unique_features_table_.emplace_back (std::move(indices_table_per_scale));
+    diff_vector.clear();
```

Comment:
Shall we move this to the top of the loop? That'll make it a pre-condition instead